apps/frontend/src/pages/root/components/user-wheel/libs/components/scores-edit-modal/scores-edit-modal.tsx

```diff
@@ -59,13 +68,25 @@ const ScoresEditModal: React.FC<Properties> = ({
 		[scores, dispatch],
 	);
 
+	const handleDiscardChanges = useCallback(() => {
+		setScores(originalScoresReference.current);
```

Comment:
No need to keep a reference. When discarding get the values from Backend

Reply:
I decided keep the data using use state. I try to call the api from backend but it has same value. I think because scores state related each others.

Reply:
Not sure to understand. We don't need to keep this data in state. When you want to discard changes, get the value from the Backend and use that

Reply:
Correct me if i wrong, did you mean call the API using the redux like this?

![dispatch1](https://github.com/user-attachments/assets/8559fa9a-e988-4821-be9f-46fd3861a60a)

Then get the scores using use app selector
![dispatch1](https://github.com/user-attachments/assets/60a18790-de30-47de-9b61-97b2c5deed5c)

Reply:
If yes, I think this can't implement because we have same state. So if we change the scores value with trigger `handleSliderChange` function it can be make all of the scores state same value and it can't be reset the value before.

Reply:
I set the scores in use effect hook and it work but I don't know that's you mean or not?

Reply:
After dispatching the scores, in state we will have saved scores from Backend. Use them to update the slider state

Reply:
I think it's already handle in use effect
![dispatch1](https://github.com/user-attachments/assets/fd5697fa-a7d9-41af-8c3c-d71c060e473e)